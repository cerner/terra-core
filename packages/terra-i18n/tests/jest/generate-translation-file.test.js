const generateTranslationFile = require('../../scripts/aggregate-translations/generate-translation-file');

describe('generate translations file', () => {
  let translationsFile;
  beforeAll(() => {
    const unsortedMessages = {
      'Terra.test.fixtures.test': 'Test...',
      'Terra.More.fixtures.test': 'Test...',
      'Terra.Fixtures1.test': 'Test...',
      'Terra.fixtures.test': 'Test...',
    };

    translationsFile = generateTranslationFile('en', unsortedMessages);
  });

  it('creates the translation file', () => {
    expect(translationsFile).toMatchSnapshot();
  });

  it('sorts the translation messages', () => {
    const sortedMessages = {
      'Terra.fixtures.test': 'Test...',
      'Terra.Fixtures1.test': 'Test...',
      'Terra.More.fixtures.test': 'Test...',
      'Terra.test.fixtures.test': 'Test...',
    };
    expect(translationsFile).toMatch(JSON.stringify(sortedMessages, null, 2));
  });
});
