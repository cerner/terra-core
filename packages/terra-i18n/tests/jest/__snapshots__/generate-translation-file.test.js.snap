// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate translations file creates the translation file 1`] = `
"import { addLocaleData } from 'react-intl';
import localeData from 'react-intl/locale-data/en';

addLocaleData(localeData);

const messages = {
  \\"Terra.fixtures.test\\": \\"Test...\\",
  \\"Terra.Fixtures1.test\\": \\"Test...\\",
  \\"Terra.More.fixtures.test\\": \\"Test...\\",
  \\"Terra.test.fixtures.test\\": \\"Test...\\"
};
const areTranslationsLoaded = true;
const locale = 'en';
export {
  areTranslationsLoaded,
  locale,
  messages
};"
`;
